### PR TITLE
Engine Developer Guide RFC

### PR DESCRIPTION
**Pull request by sptramer**
Signed-off-by: Stephen Tramer <169061+sptramer@users.noreply.github.com>

### CODE REVIEW

File: rfcs/engine-developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+## New guide: Engine Developer Guide
+
+Currently, users of Open 3D Engine (O3DE) only have one primary product manual: The User Guide (we also offer the Atom Guide, which is intended to be rolled into the User Guide at a future date.) While for most project developers and content creators the user guide is intended to be sufficient, O3DE has other audiences:
+
+* O3DE is an open-source product that is always actively seeking contributors onboarding into its code base.
+* O3DE operates in a product space where developers often need very specific low-level technical details, or guidance on performing custom implementations of the core engine.
+
+To support those considerations, there should be an Engine Developer Guide added to o3de.org. The Engine Developer Guide would cover low-level documentation for contributors and engine modification from an official, vetted, quality-controlled source.
+Motivation
```
Comment (lmbr-pip) [inline rfcs/engine-developer-guide.md:9]: This appears to be a section heading?

File: rfcs/engine-developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+While new features in the engine require RFCs, this information is not available for all parts of O3DE. In addition, header documentation and API reference have many issues in the project's current state. In particular, API reference has been a lower priority for developers. Relying on source code can be difficult for even advanced users, especially in cases where there are many abstraction layers. With the modular design of O3DE, this is a particular concern.
+
+While there are long-term solutions to some of these issues such as API reference requirements for source contribution, we need a more immediate stopgap that allows developers to share notes, reference, and other critical implementation documentation. The Engine Developer Guide is intended to serve this purpose, and carrying forward, provide authoritative information on designs and implementation.
+
+### Advantages
+
+* Provides a dedicated repository of information for low-level developers, available from the same source as user-facing documentation. Initiatives to expand existing API reference and core documentation in the future will help focus the contents of the engine developer guide.
```
Comment (lmbr-pip) [inline rfcs/engine-developer-guide.md:19]: Do we have a definition of "low-level developers"?

**Review by lmbr-pip — COMMENTED** [review]
IMHO it would be good to point at some pre-existing documentation that provides example of "engine developer" facing content.

Are we thinking along the lines of https://wiki.blender.org/wiki/Main_Page ? 

We have a lot of content that is scattered across o3de/community for example that would be good to include (code style guides)

It would be really useful to have a proposed set of sections and link to content we have.

File: rfcs/engine-developer-guide.md (hunk 2 above)
Comment (sptramer) [reply rfcs/engine-developer-guide.md:19]: Covered on Line 8:
> The Engine Developer Guide would cover low-level documentation for contributors and engine modification from an official, vetted, quality-controlled source.

File: rfcs/engine-developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+* Creates a specific dividing line between "project development" (user guide) and "engine development" (engine developer guide).
+* Allows for different review standards from sig-docs-community through processes and procedures described in #61.
```
Comment (hultonha) [inline rfcs/engine-developer-guide.md:22]: nit: Could #61 also include the full URL to the issue please

File: rfcs/engine-developer-guide.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+1. All documentation goes to the `development` branch User Guide. While a feature is in development, we may consider "users" to be alpha/beta testers who would be comfortable working with lower-quality instructions. *However*, during stabilization for a release, this would require sig-docs-community (relying on other SIGs) to evaluate content in the User Guide for release.
+1. All documentation goes to the engine developer guide. Users engaging with a prototype feature are most likely there to test, submit bug reports, or contribute improvements. *However*, there is an opposite stabilization risk here: During stabilization, sig-docs-community (relying on other SIGs) would evaluate content in the Engine Developer Guide for release.
+
+*Recommendation*: Developers who understand where to split their content should *Split at time of submission*. For developers who are unsure, they should *Submit to developer guide* and ideally request a reviewer from sig-docs-community for evaluating content for split, to create an issue to split content when the feature is stable.
+
+**Example 2 - Gem development File locations**
```
Comment (hultonha) [inline rfcs/engine-developer-guide.md:66]: uber nit: File -> file

**Comment by chanmosq** [comment]
>>  IMHO it would be good to point at some pre-existing documentation that provides example of "engine developer" facing content.
> 
> We have one public PR right now which fits perfectly into the model of the Engine Developer Guide: https://github.com/o3de/o3de.org/pull/2122

[Atom's developer guide](https://www.o3de.org/docs/atom-guide/dev-guide/) can also help kickstart it

**Review by chanmosq — APPROVED** [review]

File: rfcs/engine-developer-guide.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+```
+_index.md           # Overview page, maintained jointly by o3de/maintainers and sig-docs-community
+architecture/*      # Broad architecture documents
+framework/*         # Per-framework documentation.
+framework/_index.md # Index linking to all framework pages, with a brief overview of the framework.
+#framework/azcore   # Naming example; Frameworks should be named after their associated binary products.
+gem/*               # Per-gem documentation. Only provided for Gems included in the O3DE distribution.
+gem/_index.md       # Index linking to all Gem pages, with a brief overview of each Gem from a technical level.
+#gem/networking     # Naming example; Gems should be named after the gem name, not any produced artifacts.
```
Comment (willihay) [inline rfcs/engine-developer-guide.md:95]: Suggest we add a `tool` directory for the developer docs for included, standalone tools, such as Project Manager.
```suggestion
#gem/networking     # Naming example; Gems should be named after the gem name, not any produced artifacts.
tool/*              # Per-tool documentation for standalone tools included in the O3DE distribution.
tool/_index.md      # Index linking to all tool pages, with a brief overview of each tool.
#tool/project-manager # Naming example; Tools should be named after the tool's common program name, not necessarily the executable filename.
```

**Pull request merged**